manticore/core/smtlib/constraints.py

```diff
@@ -165,13 +167,14 @@ def declarations(self):
         for a in self.constraints:
             try:
                 declarations.visit(a)
-            except BaseException:
-                # there recursion limit exceeded problem,
-                # try a slower, iterative solution
-                #logger.info('WARNING: using iterpickle to dump recursive expression')
-                #from utils import iterpickle
-                #file('recursive.pkl', 'w').write(iterpickle.dumps(a))
-                raise
+            except RuntimeError:
```

Comment:
note; this PR includes handling that is missing in 0.2.0 for expression declarations that exceed the max recursion